index_batcher.go

```diff
@@ -0,0 +1,164 @@
+package bleve
+
+import (
+	"fmt"
+	"sync"
+	"time"
+)
+
+// NOTE XXX As this approach uses carefully orchestrated interactions
+// between a mutex, channels, and reference swapping, be sure to carefully
+// consider the impact of reordering statements or making changes to the
+// patterns present in the code.
+
+// bleveIndex is an alias that allows the IndexBatcher to embed a Index
+type bleveIndex Index
```

Comment:
I would change the comment to explain that using Index directly as an embedded field then conflicts with the Index method.  It wasn't obvious to me and I had to try it to find out.